src/client.ts

```diff
@@ -271,19 +260,26 @@ export default class Client implements ClientInterface {
 
     fs.writeFileSync(customGemfilePath, gemfile.join("\n"));
 
-    // Copy the current `Gemfile.lock` to the `.ruby-lsp` directory to make sure we're using the right versions of
-    // RuboCop and related extensions. Because we do this in every initialization, we always use the latest version of
-    // the Ruby LSP
-    if (fs.existsSync(path.join(this.workingFolder, "Gemfile.lock"))) {
-      fs.cpSync(
-        path.join(this.workingFolder, "Gemfile.lock"),
-        path.join(this.workingFolder, ".ruby-lsp", "Gemfile.lock")
-      );
-    }
+    const lastUpdatedAt: number =
+      this._context.workspaceState.get("rubyLsp.lastBundleInstall") ?? 0;
```

Comment:
What is the reasoning behind setting `lastUpdateAt` to `0` if there is nothing set at `rubyLsp.lastBundleInstall` in the workspace state? We are using this variable to see if the last update check was more than a day ago: `Date.now() - lastUpdatedAt > ONE_DAY`. When there is nothing at workspace state, `lastUpdateAt = 0`, this logic would be false and we wouldn't check for an update. Is there ever a scenario where `rubyLsp.lastBundleInstall` is just never set so we never check for an update?

Reply:
Fair point! I stopped defaulting it to zero and starting checking for `undefined` instead. If it's `undefined` (we never bundle installed), then it will do it and save something in the workspace state.